lib/framework/ensemble_widget.dart

```diff
@@ -105,7 +105,8 @@ abstract class EnsembleWidgetState<W extends EnsembleWidget> extends State<W> {
       }
 
       // To enable test mode, we need to add --dart-define="testmode=true"
-      const isTestMode = String.fromEnvironment("testmode") == "true";
+      final envString = const String.fromEnvironment("testmode").toLowerCase();
```

Comment:
environment variables should be retrieved at the load time and stored in a config object, they must not be deep in widget code. That way it's easier to understand and change how env variables are set and the rest of the code only depends on the value of the env variables and not where or how it is set. 

Please move this to EnvConfig and make EnvConfig a singleton. It should be a singleton anyway